### ISSUE TITLE
Investigate whether it's possible to run NanoAOD workflows with Dataset WorkQueue start policy

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11163]
**Impact of the new feature**
Global WorkQueue (and maybe WMAgent)

**Is your feature request related to a problem? Please describe.**
Yes, when there is a flood of NanoAOD workflows (reading in MiniAOD(SIM) and writing out NanoAOD(SIM)) workflows, the system is not able to keep up with condor job pressure because each workqueue element spawns only a handful of jobs in the agent.
 
In other words, the rate in which we complete jobs is higher than the rate to make them (GQ --> LQ --> WMBS --> Condor).

**Describe the solution you'd like**
In addition to finding out the distribution of blocks in the MINIAOD{SIM} datasets (if it's very few blocks per dataset, this work isn't worth it!), we should investigate whether it's possible to change the start WorkQueue policy for NanoAOD workflows, such that they use `Dataset` instead of `Block`, decreasing the amount of workqueue elements created in global workqueue.

If this is doable, then each NanoAOD{SIM} workflow would actually create a single workqueue element (to be replicated to the agents), however, this workqueue element would actually refer to all the blocks in the input dataset (in terms of agent WMBS operations).

**Describe alternatives you've considered**
Relax WMAgent WorkQueueManager constraints; or deploy more agents; or have specialized agents only processing a subset of the workflows.

**Additional context**
The outcome of this issue will be either:
* more issues to track and implement the necessary changes to support it in this model
* or a decision that this is actually not possible or feasible, then (re-)evaluating other ways to resolve this problem.

**Comment by amaltaro** [comment]
Using the test code provided in #11219, I managed to run a ReReco workflow that created a single workqueue element, even though there were 2 input blocks, thus properly using the `Dataset` policy. Note that change to workqueue and data policy needs to be applied both to central services and to WMAgent. 

Looking into WMAgent logs (WorkQueueManager), I found one misleading line (first one reporting 423 files):
```
2022-07-18 22:52:56,899:139989291628288:INFO:WMBSHelper:"amaltaro_ReReco_LumiMask_July2022_Val_220718_215847_7041" Injecting block /NoBPTX/Run2018D-12Nov2019_UL2018-v1/MINIAOD (423 files) into wmbs.
2022-07-18 22:52:56,899:139989291628288:INFO:WMBSHelper:Adding files into WMBS for amaltaro_ReReco_LumiMask_July2022_Val_220718_215847_7041 with PNNs: ['T2_CH_CERN', 'T1_US_FNAL_Disk']
2022-07-18 22:52:57,017:139989291628288:INFO:WMBSHelper:Inserting 3 files in bulk into WMBS for amaltaro_ReReco_LumiMask_July2022_Val_220718_215847_7041
```

but DBSBuffer did receive only the files that matched the LumiList, so we are Okay here.

Last but not least, the `Dataset` WorkQueue start policy requires all the blocks to be available in the same RSEs, otherwise there is no input data location intersection and the workflow can't run. That means we need to also instrument MSTransferor to create input rules with `grouping=ALL`, for workflows with an `InputDataset=/*/*/MINIAODSIM.`

I guess data MINIAOD is too large to be placed under a single rule/RSE, so we better keep it using the standard `Block` start policy.

In short, it's doable and apparently not much work.
